### ISSUE TITLE
batch updates unexpectedly stopped

### ISSUE THREAD

**Issue by tiancihe** [antvis/X6#929]
### Expected Behavior

batch updates should should only be stopped by `graph.stopBatch()` call once started by `graph.startBatch()` call

### Current Behavior

if there are any mutations in event handlers, batch update gets stopped before actually calling `graph.stopBatch()`

### Steps To Reproduce

https://codesandbox.io/s/charming-swartz-zmpq7?file=/src/app.tsx

1. comment out any mutation actions in event handlers, batch:stop event is emitted only after calling `graph.stopBatch()`
2. add mutation actions in event handlers, batch:stop event is emitted multiple times

**Comment by tiancihe** [comment]
maybe the batch:stop event is falsely emitted, the history undoStack is correct

**Comment by NewByVector** [comment]
I'm sorry I don't quite understand you. startBatch and stopBatch come in pairs, you can：
```ts
graph.on("batch:start", ({ name }) => console.log('start:batch', name));
graph.on("batch:stop", ({ name }) => console.log('stop:batch', name));
```

**Comment by tiancihe** [comment]
> I'm sorry I don't quite understand you. startBatch and stopBatch come in pairs, you can：
> 
> ```ts
> graph.on("batch:start", ({ name }) => console.log('start:batch', name));
> graph.on("batch:stop", ({ name }) => console.log('stop:batch', name));
> ```

So every mutation in event handlers will get a different batch session automatically , and if you start a batch session manually , these sessions' undo actions will be merged into a single batched action in history undoStack , but the session events will still be emitted nontheless ? This behaviour works fine , but i'm finding it quite confusing .

**Comment by OpportunityLiu** [comment]
![image](https://user-images.githubusercontent.com/13471233/116645375-4c061900-a9a8-11eb-8282-828239f73d9a.png)

Batch actions can be nested. If you print the names, it would be more clear.
